File: ClientApp/src/core/models/productIndexationState.ts
import Review from './review';

export default interface ProductIndexationState {
	productAsin: string
	indexationStatus: number,
	errorMessage?: string,
	
	name?: string,
	url?: string,
	by?: string,
	ratingAverage?: string,
	ratingTotalCount?: string,
	reviewTotalCount?: string,
	indexedReviews?: Review[],
}

File: Data/ProductIndexationState.cs
using System.Collections.Generic;

namespace ReviewsIndexer.Data
{
    public class ProductIndexationState
    {
        public enum Status { Waiting, Indexing, Indexed, Error }

        public ProductIndexationState(string productAsin)
        {
            ProductAsin = productAsin;
            IndexationStatus = Status.Waiting;
        }

        public string ProductAsin { get; set; }

        public string Name { get; set; }
        public string Url { get; set; }
        public string By { get; set; }

        public string RatingAverage { get; set; }
        public string RatingTotalCount { get; set; }
        public string ReviewTotalCount { get; set; }

        public Status IndexationStatus { get; set; }

        public IEnumerable<Review> IndexedReviews { get; set; }
        
        public string ErrorMessage { get; set; }
    }
}
